source/bin_expoll.cpp

```diff
@@ -1,4 +1,5 @@
 #include "common.h"
+#include "mod_mul.cpp"
```

Comment:
Nitpicking here, but including `.cpp`s directly isn't a best practice. We should at least add some `#pragma once`s to avoid multiple definitions if a file ends up being included multiple times, and change the files to be `.h`. Maybe it's worth to move these changes to a separate PR to have the new segment tree by itself